File: pilhas-filas/python/is_balanced.py
def is_balanced(text):
    stack = []
    for char in text:
        if char == '(':
            stack.append(char)
        elif char == ')':
            if len(stack) == 0:
                return False
            stack.pop()
    return len(stack) == 0


print(is_balanced("(())()"))  # true
print(is_balanced("())("))    # false

File: pilhas-filas/javascript/is-balanced.js
function isBalanced(text) {
  let stack = [];
  for (let char of text) {
    if (char === "(") {
      stack.push(char);
    } else if (char === ")") {
      if (stack.length === 0) {
        return false;
      }
      stack.pop();
    }
  }
  return stack.length === 0;
}

console.log(isBalanced("(())()")); // true
console.log(isBalanced("())(")); // false
